### PR TITLE
Quasar 17.5.0 psvamb 25152

### PR DESCRIPTION
**Pull request by hilak**

### CODE REVIEW

File: plugins/vendor/VendorPlugin.php [hunk 1]
```diff
@@ -17,8 +17,37 @@ public static function getServicesMap()
...
+		if (!$baseEnumName)
+		{
+			return array('VendorIntegrationPrepBatchJobType');
+		}
+
+		switch ($baseEnumName)
+		{
+			case 'BatchJobType':
+				return array('VendorIntegrationPrepBatchJobType');
+				break;
```
Comment (MosheMaorKaltura) [inline plugins/vendor/VendorPlugin.php:39]: remove the break

File: plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	public function getDriveItem($driveId, $driveItemId)
+	{
+		$service = $this->apiUrl .  "/v1.0/drives/$driveId/items/$driveItemId";
+		$response = $this->sendGraphRequest($service);
+
+		if ($response)
+		{
+			if (isset ($response['source']) &&
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php:69]: Use const for the strings in the API

File: plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+				$response['source']['application'] == 'Teams' &&
+				isset ($response['media']) && isset ($response['media']['mediaSource'])
+				&& isset ($response['media']['mediaSource']['contentCategory']) && $response['media']['mediaSource']['contentCategory'] == 'Meeting')
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php:72]: string comparison ===

File: plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+	public function getDriveItem($driveId, $driveItemId)
+	{
+		$service = $this->apiUrl .  "/v1.0/drives/$driveId/items/$driveItemId";
+		$response = $this->sendGraphRequest($service);
+
+		if ($response)
+		{
+			if (isset ($response['source']) &&
+				$response['source']['application'] == 'Teams' &&
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php:70]: string comparison ===

File: plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php (hunk 3 above)
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php:72]: No need to to make all this isset(..) tests, you directly check isset ($response['media']['mediaSource']['contentCategory'], no notice will be raised.

File: plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		$authHeader = "Authorization: Bearer {$this->bearerToken}";
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+		curl_setopt($ch, CURLOPT_HTTPHEADER, array($authHeader));
+		$result = curl_exec($ch);
+		$responseCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+		//TODO add handling for 427 and 503 errors.
+
+		curl_close($ch);
+
+		if ($responseCode == 200)
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftGraphClient.php:103]: Use KCurlHeaderResponse::HTTP_STATUS_OK

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+
+		$existingDropFolderFiles = $this->loadDropFolderFiles();
+
+		$files = scandir($driveLocationsDir);
+
+		$currentFile = null;
+		foreach ($files as $file)
+		{
+			if ($file == "." || $file == ".." || strpos($file,self::LOCKED_FILE_MIDFIX) || strpos($file, $driveLocationsFileNamePrefix . $this->dropFolder->id) === false)
+			{
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php:50]: use single quotes

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+
+		$newDriveUrlAssoc = array();
+		$line = fgetcsv($drivesFileHandle);
+
+		while ($line)
+		{
+			list ($userId, $driveLastPageUrl) = $line;
+			KalturaLog::info("Handling drive for user ID: $userId, drive URL: $driveLastPageUrl");
+			$items = $this->graphClient->sendGraphRequest($driveLastPageUrl);
+			foreach ($items['value'] as $item)
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php:85]: Use const for all the strings 'id' m 'value' .....

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+			return null;
+		}
+
+	}
+
+	protected function retrieveParticipants($item)
+	{
+		$participants = array();
+
+		$callRecordId = $item['source']['externalId'];
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php:187]: const

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+				break;
+		}
+
+		KBatchBase::unimpersonate();
+	}
+
+	protected function addAsNewContent (KalturaBatchJob $job, KalturaDropFolderContentProcessorJobData $data, KalturaMicrosoftTeamsDropFolder $folder)
+	{
+		/* @var $data KalturaDropFolderContentProcessorJobData */
+		$resource = $this->getIngestionResource($job, $data);
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php:252]: Need to indicate the source of this entry is TEAMS plugin by adding extension to entrySourceType in teams plugin

File: plugins/vendor/microsoft/teams/lib/api/types/KalturaMicrosoftTeamsDropFolder.php [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+			$this->clientSecret = $dbVendorIntegrationItem->getClientSecret();
+			$this->clientId = $dbVendorIntegrationItem->getClientId();
+			$this->tenantId = $dbVendorIntegrationItem->getAccountId();
+		}
+		parent::doFromObject($source_object, $responseProfile); // TODO: Change the autogenerated stub
+	}
+
+	public function toObject($dbObject = null, $skip = array())
+	{
+		if (is_null($dbObject)) {
```
Comment (MosheMaorKaltura) [inline plugins/vendor/microsoft/teams/lib/api/types/KalturaMicrosoftTeamsDropFolder.php:70]: { in new line

File: plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php (hunk 9 above)
Comment (hilak) [reply plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php:252]: @MosheMaorKaltura  according to Meirav, the same purpose can be accomplished by setting the entry's admin tags, as the Zoom plugin also does not add a new source type for entries.

**Pull request closed without merging**